### PR TITLE
Introduce version command

### PR DESCRIPTION
**Pull request by marc0der**
- First iteration of version command.
- Handle edge cases for version and bolster tests.
- Colorize version output.
- Alphabetise dependencies.

### CODE REVIEW

File: src/bin/version/main.rs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    match env::var(SDKMAN_DIR_ENV_VAR) {
+        Ok(s) => s,
+        Err(_) => fallback_sdkman_dir(),
+    }
+}
+
+fn fallback_sdkman_dir() -> String {
+    dirs::home_dir()
+        .map(|dir| dir.join(DEFAULT_SDKMAN_HOME))
+        .unwrap()
```
Comment (marc0der) [inline src/bin/version/main.rs:31]: Just using `unwrap` here because things _should_ never go wrong here (TM) :smile:

Comment (helpermethod) [reply src/bin/version/main.rs:31]: Good point! Will this also work reliably on Windows? I remember looking for a solution to find the HOME directory regardless of platform and it seemed that this was basically impossible to do reliably.

File: src/bin/version/main.rs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+fn infer_sdkman_dir() -> String {
+    match env::var(SDKMAN_DIR_ENV_VAR) {
+        Ok(s) => s,
+        Err(_) => fallback_sdkman_dir(),
+    }
+}
+
+fn fallback_sdkman_dir() -> String {
+    dirs::home_dir()
+        .map(|dir| dir.join(DEFAULT_SDKMAN_HOME))
```
Comment (helpermethod) [inline src/bin/version/main.rs:30]: If I remember correctly using `.join` on a Path in Rust was relatively expensive. Probably better to use `PathBuf` to construct `Path`s.

`dirs::home_dir` already returns a `PathBuf` btw 😄 .

File: src/bin/version/main.rs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        .to_str()
+        .unwrap()
+        .to_string()
+}
+
+fn locate_version_file(base_dir: String) -> Option<OsString> {
+    Some(
+        Path::new(base_dir.as_str())
+            .join(VERSION_FILE)
+            .into_os_string(),
```
Comment (helpermethod) [inline src/bin/version/main.rs:41]: Is the conversion actually needed? Most fs-related APIs accept a "Path-like" object.

Comment (marc0der) [reply src/bin/version/main.rs:41]: Unfortunately yes. You can't get from a `PathBuf` to a normal string, as it could be different on each OS.

File: src/bin/version/main.rs (hunk 2 above)
Comment (marc0der) [reply src/bin/version/main.rs:30]: That's right, I'm mapping over the `Option` that `dirs::home_dir()` returns and joining `.sdkman` onto the `PathBuf`.

Comment (marc0der) [reply src/bin/version/main.rs:29]: All cut over to `PathBuf` now.
